### PR TITLE
Connection: increase timeout on the token request

### PR DESCRIPTION
**Pull request by kbrown9**
#### Changes proposed in this Pull Request:
* We're seeing occasional timeouts on the token request during user authorization. Let's try increasing the timeout from the default 5s to 30s.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This changes an existing part of Jetpack.

#### Testing instructions:
The test site must have Jetpack activated and disconnected.

1. Verify that the you can successfully connect to WPCOM.

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

**Comment by dereksmart** [comment]
So this should work in solving a lot of cases. 

I've tested it by adding a `sleep( 6 )` call in wpcom's Jetpack_Server_Version->token() method. It definitely works to extend the timeout. 

One side effect, is that this hard-coded change bypasses core's `http_request_timeout` filter. I tested this by manually setting it like so in a functionality plugin on my site. I'm not saying this is necessarily bad, but that it should be an explicit decision. 

```add_filter( 'http_request_timeout', 'modify_timeout' );
function modify_timeout() {
	return 1;
}
```

While watching the requests going through `WP_HTTP->request()`, I did notice that we're being incredibly generous with the call to `jetpack.register`, [setting a timeout of 30 seconds](https://github.com/Automattic/jetpack/blob/master/packages/connection/src/class-manager.php#L820)! 

```[hooks] => WP_HTTP_Requests_Hooks Object
        (
            [url:protected] => https://jetpack.wordpress.com/jetpack.register/1/
            [request:protected] => Array
                (
                    [method] => POST
                    [timeout] => 30
                    [redirection] => 5
```

This override is also not respecting core's filter, and reading the code it looks to definitely be on purpose. 

This may actually suggest that we are in the clear to bump this timeout even more. 

Thoughts?

**Comment by kbrown9** [comment]
> So this should work in solving a lot of cases.

Yes - I'm looking forward to seeing the results!

> One side effect, is that this hard-coded change bypasses core's http_request_timeout filter. I > tested this by manually setting it like so in a functionality plugin on my site. I'm not saying this > is necessarily bad, but that it should be an explicit decision.

Good point! We should use the filter to set the timeout so we have the option of overriding it if necessary. I'll submit another commit to make this change.

> While watching the requests going through WP_HTTP->request(), I did notice that we're being > incredibly generous with the call to jetpack.register, setting a timeout of 30 seconds!
>
> This may actually suggest that we are in the clear to bump this timeout even more.

The token request happens within the authorization request from WPCOM, so we probably don’t want to go overboard with the timeout increase. The connection process will fail if the authorization request or the token request time out, but I think we want to be able to determine which request caused the failure. I'll try 30s in the next commit.

File: packages/connection/src/class-manager.php [hunk 1]
```diff
@@ -1590,7 +1590,9 @@ public function get_token( $data ) {
 			),
 		);
 
+		add_filter( 'http_request_timeout', array( $this, 'increase_timeout' ) );
```
Comment (kraftbj) [inline packages/connection/src/class-manager.php:1593]: Let's bump the priority on this filter to something high -- say `9999` or even `PHP_INT_MAX - 1`-- so we can still change it, but the average filter won't hit it.

Generally, we don't want other plugins to be modifying this without intent.

Comment (kbrown9) [reply packages/connection/src/class-manager.php]: Good point! I changed the priority to PHP_INT_MAX-1 in commit 5011a63.

**Comment by kbrown9** [comment]
> Just a quick note should we change the PR description (Let's try increasing the timeout from the default 5s to 10s.) to 30 which matches the code?

I fixed it - thanks!

**Pull request merged**